### PR TITLE
feat(web): make app controlable via web requests

### PR DESCRIPTION
**Pull request by lauevrar77**

### CODE REVIEW

**Comment by lauevrar77** [comment]
@MrWazaby Still a WIP. I will warn you when review will be required

**Comment by lauevrar77** [comment]
@MrWazaby Ready to be reviewed

File: README.md [hunk 1]
```diff
@@ -1 +1,159 @@
...
+The main features of Maestro are :
+
+* Agnostic of the client and worker language
+* Timeout
+* Retries
+* Cancel
+* Cron Tasks
+* Oblivious : No task payload or result is saved permanently. Your data is safe.
+
+## Roadmap
```
Comment (MrWazaby) [inline README.md]: You can move this to a dedicated roadmap.md file.

File: README.md [hunk 2]
```diff
@@ -1 +1,159 @@
...
+
+* Tasks can be created (with payload)
+* Tasks can be got by workers from queue
+* Workers can change complete (with result), fail or timeout tasks
+* Retries are working
+* Timeout is working
+* Cron tasks are **working**
+* Cancel is working
+
+## Run
```
Comment (MrWazaby) [inline README.md:22]: Maybe we could add the docker pull command (to pull image from the Docker Hub).

File: cmd/srv/main.go [hunk 3]
```diff
@@ -2,24 +2,118 @@ package main
...
-	repo := goddd.NewExentStoreRepository("http://localhost:4000", eventPublisher)
-	taskService := services.NewTaskService(&repo)
-	taskID := taskService.Create("test", 10, 1, nil)
-	fmt.Println(taskService.Select(taskID))
+	taskStateProjection := projection.NewTaskStateProjection(mongoDB)
+	payloadRepo := repository.NewPayloadRepository(redisClient)
+	eventPublisher.Register(taskStateProjection)
+	taskFinishedListener := listener.NewTaskFinishedListener(payloadRepo)
+	eventPublisher.Register(taskFinishedListener)
+	taskRepo := goddd.NewExentStoreRepository("http://localhost:4000", &eventPublisher)
```
Comment (MrWazaby) [inline cmd/srv/main.go]: You should add this value in a variable.

File: .gitignore [hunk 4]
```diff
@@ -1 +1,2 @@
-.env
\ No newline at end of file
+.env
```
Comment (MrWazaby) [inline .gitignore:1]: You should add in the readme that the .env is needed.

File: cmd/srv/main.go [hunk 5]
```diff
@@ -2,24 +2,118 @@ package main
...
+	)
+	queueNextTaskHandler := httptransport.NewServer(
+		endpoint.QueueNextEndpoint(taskService, &taskStateView, payloadView),
+		rest.DecodeQueueNextRequest,
+		rest.EncodeJSONResponse,
+	)
+
+	router := httprouter.New()
+	router.Handler("POST", "/api/task/create", createTaskHandler)
+	router.Handler("POST", "/api/task/get", taskStateHandler)
```
Comment (MrWazaby) [inline cmd/srv/main.go:73]: Why using POST and not GET here?

File: cmd/srv/main.go [hunk 6]
```diff
@@ -2,24 +2,118 @@ package main
...
+	router.Handler("POST", "/api/task/complete", completeTaskHandler)
+	router.Handler("POST", "/api/task/fail", failTaskHandler)
+	router.Handler("POST", "/api/task/timeout", timeoutTaskHandler)
+	router.Handler("POST", "/api/queue/next", queueNextTaskHandler)
```
Comment (MrWazaby) [inline cmd/srv/main.go:77]: Why using POST and not GET here?

File: cmd/srv/main.go (hunk 5 above)
Comment (lauevrar77) [reply cmd/srv/main.go:73]: It's true that it is not very REST compliant but it is a convention in Owlint Backend Services. No URL parameters. All parameters are passed explicitly in JSON body.

**Pull request merged**